src/main/java/com/google/devtools/build/lib/remote/RemoteCache.java

```diff
@@ -260,6 +262,36 @@ private static Path toTmpDownloadPath(Path actualPath) {
     return actualPath.getParentDirectory().getRelative(actualPath.getBaseName() + ".tmp");
   }
 
+  /**
+   * Exception which represents a collection of IOExceptions for the purpose
+   * of distinguishing remote communication exceptions from those which occur
+   * on filesystems locally. This exception serves as a trace point for the actual
+   * download, so that the intented operation can be observed in a stack, with all
+   * constituent exceptions available for observation.
+   */
+  static class DownloadException extends Exception {
+    // true since no empty DownloadException is ever thrown
+    boolean allCacheNotFoundException = true;
+
+    DownloadException() {
+    }
+
+    DownloadException(IOException e) {
+      add(e);
+    }
+
+    void add(IOException e) {
```

Comment:
Why not override addSuppressed?

Reply:
I wanted a filter on this, so that it couldn't contain/suppress anything but IOExceptions. That is probably unnecessary, but I felt it needed scope.

Reply:
And as it turns out, addSuppressed is `final`